release-common-lib/src/main/java/org/reactome/util/compare/AttributeProcessor.java

```diff
@@ -0,0 +1,171 @@
+package org.reactome.util.compare;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.gk.model.GKInstance;
+import org.gk.model.Instance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.schema.SchemaAttribute;
+import org.gk.schema.SchemaClass;
+
+/**
+ * This class provides helper methods in processing, manipulating and obtaining attribute objects
+ * for the DBObjectComparer class
+ * @author jweiser
+ */
+class AttributeProcessor
+{
+	private static Map<String, Map<AttributeRelationshipType, List<SchemaAttribute>>>
+		schemaClassToAttributesMap = new ConcurrentHashMap<>();
+
+	/**
+	 * Returns the attributes associated, via the passed attribute relationship type, with the passed schema class.
+	 * @param schemaClass SchemaClass for which to retrieve attributes
+	 * @param attributeRelationshipType Type of attributes to retrieve from the passed schema class (i.e. 'regular' or
+	 * referrer attributes)
+	 * @return List of SchemaAttribute objects associated, via the passed attribute relationship type, with the passed
+	 * schema class
+	 */
+	@SuppressWarnings("unchecked")
+	static List<SchemaAttribute> getAttributes(
+		SchemaClass schemaClass, AttributeRelationshipType attributeRelationshipType
```

Comment:
Is `AttributeRelationshipType` an external class or a Reactome one? I've never seen it before

Reply:
I'm pretty sure that's an `enum` Joel created to make it easier to check relationship types. https://github.com/reactome/data-release-pipeline/pull/121/commits/6b9cecd404784e334488c24488846abab86f9e2f#diff-b4b21932944d31bdd5550b507e7e2195R723